abz/acousticbrainz.py

```diff
@@ -93,7 +93,18 @@ def submit_features(recordingid, features):
     featstr = json.dumps(features)
 
     host = config.settings["host"]
-    url = compat.urlunparse(('http', host, '/%s/low-level' % recordingid, '', '', ''))
+    if sys.version_info < (3, 0):
```

Comment:
This only has to happen once, it should be at the beginning of the file instead of in the submit method.

Reply:
Is this for injecting SNI support? If so, It should be `< (3, 4)`, and maybe even not needed for `>= (2, 7 ,9) and < (3)`. (To make the code simpler, it could also just be decided to drop support for versions older than that and specify this in `setup.py` (see #50). If so, I'd recommend making a release with #50 merged first in case there *are* people out there using ancient Python versions—they'd still be able to `pip install` and get a version.)

Reply:
good point about the 3.x versions that need it. Thanks. 
You're right about the fixes in 2.7.9+, We should have a look at what version of python are supported in common operating systems, and perhaps this will prevent us from having to do this. I see that ubuntu 14.04 LTS goes out of support this month. 16.04 LTS has 2.7.11 already. I'm not sure about OSX/macOS versions - we should double-check this going back 3 or 4 versions.
Python 2.7.9 is 4 years old: https://www.python.org/downloads/release/python-279/
I'm tempted to follow your suggestion and set this as the lowest support version.